File: .cursor-agent/hooks/afterMCPExecution.py
#!/usr/bin/env python3
"""
afterMCPExecution Hook  
在 Agent 执行 MCP 工具后触发
"""

import sys
from pathlib import Path

sys.path.insert(0, str(Path(__file__).parent.parent / "lib"))

from agent_hook_handler import AuditHook


class AfterMCPExecutionHook(AuditHook):
    """MCP 工具执行后的审计"""
    
    def __init__(self):
        super().__init__("afterMCPExecution")
    
    def audit(self) -> None:
        """审计 MCP 工具执行"""
        tool_name = self.input_data.get("tool_name", "")
        result_json = self.input_data.get("result_json", "")
        
        if not tool_name:
            return
        
        # 检查结果中的错误
        has_error = '"error"' in result_json or '"success": false' in result_json
        
        if has_error:
            self.send_to_ortensia(
                f"工具失败：{tool_name}",
                emotion="sad"
            )
        else:
            # 成功 - 所有工具都通知
            self.send_to_ortensia(
                f"工具完成：{tool_name}",
                emotion="happy"
            )


if __name__ == "__main__":
    hook = AfterMCPExecutionHook()
    sys.exit(hook.run())

File: cursor-hooks/hooks-node/afterMCPExecution.js
#!/usr/bin/env node
/**
 * afterMCPExecution (Node)
 */

const path = require("path");
const { AuditHook } = require(path.join(__dirname, "..", "lib-node", "agent_hook_handler"));

class AfterMCPExecutionHook extends AuditHook {
  constructor() {
    super("afterMCPExecution");
  }

  audit() {
    const toolName = this.inputData.tool_name || "";
    const resultJson = this.inputData.result_json || "";
    if (!toolName) return;

    const hasError = String(resultJson).includes('"error"') || String(resultJson).includes('"success": false');
    if (hasError) {
      this.sendToOrtensia(`工具失败：${toolName}`, "sad").catch(() => {});
    } else {
      this.sendToOrtensia(`工具完成：${toolName}`, "happy").catch(() => {});
    }
  }
}

process.exit(new AfterMCPExecutionHook().run());

